### PR TITLE
Podspecs for CoreInternal and AppCheckInterop

### PR DESCRIPTION
**Pull request by paulb777**
We need Interop pods again now that we want to depend on Interop from Swift pods

End-to-end with AppCheck here and will follow on with other Interop pods in subsequent PR(s)

#no-changelogs

### CODE REVIEW

File: Firestore/Example/Benchmarks/remote_document_cache_benchmark.mm [hunk 1]
```diff
@@ -15,7 +15,6 @@
  */
 
 #import <FirebaseFirestore/FirebaseFirestore.h>
-#import "FirebaseCore/Internal/FirebaseCoreInternal.h"
```
Comment (ryanwilson) [inline Firestore/Example/Benchmarks/remote_document_cache_benchmark.mm:18]: Should we still import `FirebaseApp.h` at least? Seems like we're relying on an implicit dependency now.

File: Firestore/Source/API/FSTFirestoreComponent.mm [hunk 2]
```diff
@@ -21,7 +21,13 @@
...
 
 #import "FirebaseAppCheck/Interop/FIRAppCheckInterop.h"
-#import "FirebaseCore/Internal/FirebaseCoreInternal.h"
+#import "FirebaseCore/Internal/FIRAppInternal.h"
+#import "FirebaseCore/Internal/FIRComponent.h"
+#import "FirebaseCore/Internal/FIRComponentContainer.h"
+#import "FirebaseCore/Internal/FIRComponentType.h"
+#import "FirebaseCore/Internal/FIRDependency.h"
+#import "FirebaseCore/Internal/FIRLibrary.h"
+#import "FirebaseCore/Internal/FIROptionsInternal.h"
```
Comment (ryanwilson) [inline Firestore/Source/API/FSTFirestoreComponent.mm:30]: Does it make sense to still have some sort of umbrella header for internal APIs?

File: Gemfile [hunk 3]
```diff
@@ -15,3 +15,5 @@ source 'https://rubygems.org'
 gem 'cocoapods', '1.11.2'
 gem 'cocoapods-generate', '2.0.1'
 gem 'danger', '6.1.0'
+
+# force building -delete me
```
Comment (ryanwilson) [inline Gemfile:19]: Delete me :)

File: .github/workflows/cocoapods-integration.yml [hunk 4]
```diff
@@ -18,7 +18,7 @@ jobs:
     runs-on: macos-11
     strategy:
       matrix:
-        config: [Cocoapods_multiprojects_frameworks, Cocoapods_multiprojects_staticLibs]
+        config: [Cocoapods_multiprojects_frameworks]
```
[the remark this answers is not in the document]
Comment (paulb777) [reply .github/workflows/cocoapods-integration.yml:21]: Done. Also deleted the test config.

File: FirebaseCoreInternal.podspec [hunk 5]
```diff
@@ -0,0 +1,29 @@
+Pod::Spec.new do |s|
+  s.name             = 'FirebaseCoreInternal'
```
[the remark this answers is not in the document]
Comment (paulb777) [reply FirebaseCoreInternal.podspec:2]: That would combine all the headers into the single FirebaseCore module

File: FirebaseFunctionsSwift.podspec [hunk 6]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name                    = 'FirebaseFunctionsSwift'
-  s.version                 = '8.12.0-beta'
+  s.version                 = '9.0.0'
```
[the remark this answers is not in the document]
Comment (paulb777) [reply FirebaseFunctionsSwift.podspec:3]: Done.

File: Firestore/Example/Benchmarks/remote_document_cache_benchmark.mm (hunk 1 above)
Comment (paulb777) [reply Firestore/Example/Benchmarks/remote_document_cache_benchmark.mm:18]: Good catch!

File: Firestore/Source/API/FSTFirestoreComponent.mm (hunk 2 above)
Comment (paulb777) [reply Firestore/Source/API/FSTFirestoreComponent.mm:30]: I'm not sure it's worth it only for Firestore?  The future is modules and Swift

File: SymbolCollisionTest/Podfile [hunk 7]
```diff
@@ -36,7 +35,7 @@ target 'SymbolCollisionTest' do
 
   # Other Google Pods
 #    pod 'ARCore' Conflicts with FirebasePerformance via Clearcut and google-cast-sdk
-#    pod 'Blockly' This is a Swift Pod and requires usage of use_frameworks!
+#    pod 'Blockly'  `Blockly` does not specify a Swift version and none of the targets
```
[the remark this answers is not in the document]
Comment (paulb777) [reply SymbolCollisionTest/Podfile:38]: fixed

File: Gemfile (hunk 3 above)
Comment (paulb777) [reply Gemfile:19]: done.

File: FirebaseCoreInternal.podspec (hunk 5 above)
[the remark this answers is not in the document]
Comment (ryanwilson) [reply FirebaseCoreInternal.podspec:2]: Ahh yep, wasn't thinking. Thanks!

**Review by ryanwilson — APPROVED** [review]

**Pull request merged**